CMakeLists.txt

```diff
@@ -85,6 +85,19 @@ else ()
 endif ()
 
 ###########################################################################
+#       CCache configuration
+###########################################################################
+
+find_program(CCACHE_FOUND ccache)
+if(CCACHE_FOUND)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
+    if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Qunused-arguments -fcolor-diagnostics")
```

Comment:
Actually, it works.
This blog post explains those flags well http://petereisentraut.blogspot.nl/2011/05/ccache-and-clang.html
clang *may* generate unused variable variable warnings due to separate preprocessor and compiler steps, but it's not the case.
-fcolor-diagnostics only required if you want a colored output.

Can you please make a separate pull request for ccache with the following code (I also added space before bracket to keep code style):
```
###########################################################################
#       CCache configuration
###########################################################################
find_program (CCACHE_FOUND ccache)
if (CCACHE_FOUND)
    message (STATUS "Enable ccache")
    set_property (GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
    set_property (GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
endif (CCACHE_FOUND)```